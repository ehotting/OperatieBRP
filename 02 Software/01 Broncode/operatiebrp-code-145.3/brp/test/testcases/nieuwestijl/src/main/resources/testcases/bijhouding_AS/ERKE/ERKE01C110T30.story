Meta:
@status                 Klaar
@usecase                UCS-BY.0.AS

Narrative: R2644 Bij beëindigde staatloosheid is registratie nationaliteit uitsluitend toegestaan als de DAG daarvan op of na de DEG van staatloos ligt

Scenario:   beëindigde staatloosheid en registratie nationaliteit DAG eerder dan staatloos DEG
            LT: ERKE01C110T30

Given alle personen zijn verwijderd

!-- Init. vulling van het kind met een materieel beëindigde staatloosheid en een OUWKIG
Given enkel initiele vulling uit bestand /LO3PL-AS/LO3PL-ERKE/ERKE01C110T30-001.xls

!-- Maak van de vorige NOUWKIG een kind zodat er niet teveel ouders zijn.
Given de database is aangepast met: update kern.betr set rol=1 where pers=(select id from kern.pers where voornamen='verwijder')

!-- Init. vulling van de erkennende NOUWKIG
Given enkel initiele vulling uit bestand /LO3PL-AS/LO3PL-ERKE/ERKE01C110T30-002.xls

!-- Controleer dat staatloos materieel is beëindigd
Then in kern heeft select dateindegel,
                          waarde
                   from   kern.his_persindicatie
		   where  persindicatie in (
		                               select id
					       from   kern.persindicatie
					       where  srt in  (
					                          select id
								  from   kern.srtindicatie
							          where  id=7
							      )
					       and    pers in (
					                          select id
								  from   kern.pers
								  where  voornamen='Kind'
						              )
					   ) de volgende gegevens:
| veld        | waarde   |
| dateindegel | 20160102 |
| waarde      | true     |

Then heeft $NOUWKIG_BSN$ de waarde van de volgende query: select bsn from kern.pers where voornamen = 'Vader'
Then heeft $KIND_BSN$ de waarde van de volgende query: select bsn from kern.pers where voornamen = 'Kind'

Then heeft $RELATIE_ID$ de waarde van de volgende query: select r.id from kern.betr b join kern.relatie r on r.id = b.relatie join kern.pers p on b.pers = p.id where p.voornamen='Kind' and r.srt=3
Then heeft $KIND_BETROKKENHEID$ de waarde van de volgende query: select b.id from kern.betr b join kern.relatie r on r.id = b.relatie join kern.pers p on b.pers = p.id where p.voornamen='Kind' and r.srt=3

!-- Erkenning van het kind
When voer een bijhouding uit ERKE01C110T30.xml namens partij 'Gemeente BRP 1'

Then heeft het antwoordbericht verwerking Foutief

Then is het antwoordbericht gelijk aan /testcases/bijhouding_AS/ERKE/expected/ERKE01C110T30.xml voor expressie //brp:bhg_afsRegistreerErkenning_R