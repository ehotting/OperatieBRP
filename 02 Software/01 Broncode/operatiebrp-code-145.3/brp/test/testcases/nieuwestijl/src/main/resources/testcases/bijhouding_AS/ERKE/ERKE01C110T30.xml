<brp:bhg_afsRegistreerErkenning xmlns:brp="http://www.bzk.nl/brp/brp0200">
    <brp:stuurgegevens brp:communicatieID="stuurgegevens">
        <brp:zendendePartij>507013</brp:zendendePartij>
        <brp:zendendeSysteem>BRP</brp:zendendeSysteem>
        <brp:referentienummer>00000000-0000-0000-0042-200000000104</brp:referentienummer>
        <brp:tijdstipVerzending>2016-01-01T15:16:40.829+02:00</brp:tijdstipVerzending>
    </brp:stuurgegevens>
    <brp:parameters brp:communicatieID="CI_parameters">
        <brp:verwerkingswijze>Bijhouding</brp:verwerkingswijze>
    </brp:parameters>
    <brp:erkenning brp:objecttype="AdministratieveHandeling" brp:communicatieID="CI_adm_hand">
        <brp:partijCode>507013</brp:partijCode>
		<brp:toelichtingOntlening>er is geen toelichting</brp:toelichtingOntlening>
        <brp:gedeblokkeerdeMeldingen>
            <brp:gedeblokkeerdeMelding brp:communicatieID="CI_deblok" brp:objecttype="GedeblokkeerdeMelding" brp:referentieID="CI_adm_hand">
                <brp:regelCode>R2493</brp:regelCode>
            </brp:gedeblokkeerdeMelding>
        </brp:gedeblokkeerdeMeldingen>
		<brp:bronnen>
            <brp:bron brp:objecttype="AdministratieveHandelingBron" brp:communicatieID="CI_ah_bron_1">
                <brp:document brp:objecttype="Document" brp:communicatieID="CI_document_1">
                    <brp:soortNaam>Akte van erkenning</brp:soortNaam>
                    <brp:aktenummer>1AA0001</brp:aktenummer>
                    <brp:partijCode>507013</brp:partijCode>
                </brp:document>
            </brp:bron>
        </brp:bronnen>
        <brp:acties>
            <brp:registratieOuder brp:objecttype="Actie" brp:communicatieID="CI_actie_1">
                <brp:datumAanvangGeldigheid>2016-01-01</brp:datumAanvangGeldigheid>
                <brp:bronnen>
                    <brp:bron brp:objecttype="ActieBron" brp:referentieID="CI_ah_bron_1" brp:communicatieID="CI_bron_1"/>
                </brp:bronnen>
                <brp:persoon brp:objecttype="Persoon" brp:objectSleutel="$KIND_BSN$" brp:communicatieID="CI_kind">
                    <brp:betrokkenheden>
                        <brp:kind brp:objecttype="Betrokkenheid" brp:objectSleutel="$KIND_BETROKKENHEID$" brp:communicatieID="CI_kind_betrokkenheid">
                            <brp:familierechtelijkeBetrekking brp:objecttype="Relatie" brp:objectSleutel="$RELATIE_ID$" brp:communicatieID="CI_kind_fbr">
                                <brp:betrokkenheden>
                                    <brp:ouder brp:objecttype="Betrokkenheid" brp:communicatieID="CI_ouder_betrokkenheid">
                                        <brp:persoon brp:objecttype="Persoon" brp:objectSleutel="$NOUWKIG_BSN$" brp:communicatieID="CI_ouder"/>
                                    </brp:ouder>
                                </brp:betrokkenheden>
                            </brp:familierechtelijkeBetrekking>
                        </brp:kind>
                    </brp:betrokkenheden>
                </brp:persoon>
            </brp:registratieOuder>
			<brp:registratieNationaliteit brp:objecttype="Actie" brp:communicatieID="CI_regnati_1">
                <brp:datumAanvangGeldigheid>2016-01-01</brp:datumAanvangGeldigheid>
                <brp:persoon brp:objecttype="Persoon" brp:objectSleutel="$KIND_BSN$" brp:communicatieID="CI_regnati.pers_1">
                    <brp:nationaliteiten>
                        <brp:nationaliteit brp:objecttype="PersoonNationaliteit" brp:communicatieID="CI_regnati.nati_1">
                            <brp:nationaliteitCode>0027</brp:nationaliteitCode>
                        </brp:nationaliteit>
                    </brp:nationaliteiten>
                </brp:persoon>
            </brp:registratieNationaliteit>
        </brp:acties>
    </brp:erkenning>
</brp:bhg_afsRegistreerErkenning>
