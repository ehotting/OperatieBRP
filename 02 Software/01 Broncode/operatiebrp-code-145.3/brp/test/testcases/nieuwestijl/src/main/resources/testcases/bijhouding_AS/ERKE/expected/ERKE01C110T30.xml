<brp:bhg_afsRegistreerErkenning_R xmlns:brp="http://www.bzk.nl/brp/brp0200">
    <brp:stuurgegevens>
        <brp:zendendePartij>199901</brp:zendendePartij>
        <brp:zendendeSysteem>BRP</brp:zendendeSysteem>
        <brp:referentienummer>*</brp:referentienummer>
        <brp:crossReferentienummer>00000000-0000-0000-0042-200000000104</brp:crossReferentienummer>
        <brp:tijdstipVerzending>*</brp:tijdstipVerzending>
    </brp:stuurgegevens>
    <brp:resultaat>
        <brp:verwerking>Foutief</brp:verwerking>
        <brp:hoogsteMeldingsniveau>Fout</brp:hoogsteMeldingsniveau>
    </brp:resultaat>
    <brp:meldingen>
        <brp:melding brp:objecttype="Melding" brp:referentieID="CI_regnati_1">
            <brp:regelCode>R2644</brp:regelCode>
            <brp:soortNaam>Fout</brp:soortNaam>
            <brp:melding>*</brp:melding>
        </brp:melding>
    </brp:meldingen>
    <brp:erkenning brp:objecttype="AdministratieveHandeling">
        <brp:partijCode>507013</brp:partijCode>
        <brp:tijdstipRegistratie>*</brp:tijdstipRegistratie>
        <brp:gedeblokkeerdeMeldingen>
            <brp:gedeblokkeerdeMelding brp:objecttype="GedeblokkeerdeMelding" brp:referentieID="CI_deblok">
                <brp:regelCode>R2493</brp:regelCode>
                <brp:melding>*</brp:melding>
            </brp:gedeblokkeerdeMelding>
        </brp:gedeblokkeerdeMeldingen>
    </brp:erkenning>
</brp:bhg_afsRegistreerErkenning_R>
